lib/rust/parser/tests/parse.rs

```diff
@@ -471,6 +471,137 @@ fn import() {
+            (Export
+             ((Ident from) (Ident Foo))
+             ()
+             ((Ident export) (Ident all))
+             ()
+             ((Ident hiding) (OprApp (Ident Bar) (Ok ",") (Ident Baz))))]),
+        ("from Foo as Bar export Baz, Quux", block![
+            (Export
+             ((Ident from) (Ident Foo))
+             ((Ident as) (Ident Bar))
+             ((Ident export) (OprApp (Ident Baz) (Ok ",") (Ident Quux)))
+             () ())
+        ]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+
+// === Metadata ===
+
+
+#[test]
+fn metadata_raw() {
+    let code = [
+        "4",
+        "#### METADATA ####",
+        r#"[[{"index":{"value":7},"size":{"value":8}},"5bad897e-099b-4b00-9348-64092636746d"]]"#,
+    ];
+    let code = code.join("\n");
+    let (_meta, code) = enso_parser::metadata::parse(&code).unwrap();
+    let expected = block![
+        (Number 4)
+        ()
+    ];
+    test(code, expected);
+}
+
+#[test]
+fn metadata_parsing() {
+    let code = metadata::ORDERS_WITH_METADATA;
+    let (meta, code) = enso_parser::metadata::parse(code).unwrap();
+    let _ast = enso_parser::Parser::new().run(code);
+    let _meta: enso_parser::metadata::Metadata = meta.unwrap();
+}
+
+
+// === Type annotations and signatures ===
+
+#[test]
+fn type_signatures() {
+    let cases = [
+        ("val : Bool", block![(TypeSignature val ":" (Ident Bool))]),
+        ("val : List Int", block![(TypeSignature val ":" (App (Ident List) (Ident Int)))]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+#[test]
+fn type_annotations() {
+    #[rustfmt::skip]
+    let cases = [
+        ("val = 123 : Int", block![
+            (Assignment (Ident val) "=" (TypeAnnotated (Number 123) ":" (Ident Int)))]),
+        ("val = foo (123 : Int)", block![
+            (Assignment (Ident val) "="
+             (App (Ident foo)
+              (Group "(" (TypeAnnotated (Number 123) ":" (Ident Int)) ")")))]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+
+// === Text Literals ===
+
+#[test]
+fn inline_text_literals() {
+    #[rustfmt::skip]
+    let cases = [
+        (r#""I'm an inline raw text!""#, block![
+            (TextLiteral "\"" #("I'm an inline raw text!") "\"")]),
+        (r#"zero_length = """#, block![
+            (Assignment (Ident zero_length) "=" (TextLiteral "\"" #() "\""))]),
+        (r#"unclosed = ""#, block![(Assignment (Ident unclosed) "=" (TextLiteral "\"" #() ()))]),
+        (r#"unclosed = "a"#, block![
+            (Assignment (Ident unclosed) "=" (TextLiteral "\"" #("a") ()))]),
+        (r#"'Other quote type'"#, block![(TextLiteral "'" #("Other quote type") "'")]),
+    ];
+    cases.into_iter().for_each(|(code, expected)| test(code, expected));
+}
+
+#[test]
+fn multiline_text_literals() {
+    test("'''", block![(TextLiteral "'''" #() ())]);
+    const CODE: &str = r#"'''
```

Comment:
I guess I'll have a fun to rewrite these text literals from Rust to Java while keeping 1:1 character representation...